chain/chain/src/chain.rs

```diff
@@ -582,8 +582,41 @@ impl Chain {
                 }
             }
             chain_store_update.update_tail(height);
+            chain_store_update.commit()?;
         }
-        chain_store_update.commit()?;
+        Ok(())
+    }
+
+    pub fn clear_forks_data(&mut self, trie: Arc<Trie>, height: BlockHeight) -> Result<(), Error> {
+        if let Ok(blocks_current_height) = self.store.get_all_block_hashes_by_height(height) {
+            let blocks_current_height =
+                blocks_current_height.values().flatten().cloned().collect::<Vec<_>>();
+            for block_hash in blocks_current_height.iter() {
+                let mut current_hash = *block_hash;
+                loop {
+                    // Block `block_hash` is not on the Canonical Chain
+                    // because shorter chain cannot be Canonical one
+                    // and it may be safely deleted
+                    // and all its ancestors while there are no other sibling blocks rely on it.
+                    let mut chain_store_update = self.store.store_update();
+                    if *chain_store_update.get_block_refcount(&current_hash)? == 0 {
```

Comment:
here we can do `self.get_block_refcount` and create `chain_store_update` inside

Reply:
`get_block_refcount` in ChainStore and ChainStoreUpdate are different

Reply:
I see, `ChainStore` turns option into `DBNotFoundErr` and `ChainStoreUpdate` turns `DBNotFoundErr` into 0.
Let's keep `ChainStoreUpdate` for simplicity.